sdk/python/kfp/compiler/compiler.py

```diff
@@ -508,7 +508,7 @@ def _convert_k8s_obj_to_dic(self, obj):
     Returns: The serialized form of data.
     """
 
-    from six import text_type, integer_types
+    from six import text_type, integer_types, iteritems
```

Comment:
Is this a style change? This function is copied directly from the official Kubernetes client library.

Reply:
This is the bug.  I hit this issue because I tried to pass a dict into `op.add_env_variable`.  Line [541](https://github.com/kubeflow/pipelines/pull/411/files/e574ff660f341cf087f0ac476aad4ae421c23dc0#diff-618bb1fbca23caf0b3edc26f1b38caa8L541) accesses `iteritems` but it was only defined in a conditional branch so it threw an "object accessed before assignment" error.

Reply:
Another fix here may be to throw a ValueError if the env_variable isn't a k8s object, but It would be nice to be able to pass in a dict without having to use the k8s wrapper.  Regardless, this is definitely a bug.